backend/src/controllers/user.rs

```diff
@@ -121,29 +121,12 @@ pub async fn delete_user(
     Extension(pool): Extension<PgPool>,
 ) -> Result<(StatusCode, Json<Value>), CustomError> {
 
-    let user_model = sqlx::query_as!(UserModel, "SELECT * FROM fluke_user WHERE id = $1", id)
-        .fetch_one(&pool)
-        .await
-        .map_err(|_| CustomError::UserNotFound(id.to_string()))?;
-
-    let user_id_str = user_model.id.to_string();
-
-    let sql = "DELETE FROM fluke_user WHERE id = $1";
-
-    let _ = sqlx::query(sql)
-        .bind(id)
-        .execute(&pool)
-        .await
-        .map_err(|_| CustomError::UserNotFound(id.to_string()))?;
-
-    let _ = sqlx::query!(
-        "UPDATE message SET user_id = NULL WHERE user_id = $1",
-        id
-    )
-    .execute(&pool)
-    .await;
+    let _ = sqlx::query_as!(UserModel,"UPDATE fluke_user SET deleted=true WHERE id=($1)", id)
+    .fetch_one(&pool)
+    .await
+    .map_err(|_| CustomError::UserNotFound(id.to_string()))?;
 
-    Ok((StatusCode::OK, Json(json!({"message": "User deleted", "user_id": user_id_str}))))
+    Ok((StatusCode::OK, Json(json!({"message": "User soft-deleted", "user_id": id.to_string()}))))
```

Comment:
```suggestion
    Ok((StatusCode::OK, Json(json!({"message": "User deleted", "user_id": id.to_string()}))))
```
They don't need to know its soft